src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0]
+    ];
+
+    // reqBody is the request body, a json object
+    constructor(reqBody:any){
+        // A-J and 1-10
+        this.ShipPositions = (reqBody.ShipPositions)?(reqBody.ShipPositions):[];
+        this.MyShots = (reqBody.MyShots)?(reqBody.MyShots):[];
+        this.OpponentsShots = (reqBody.OpponentsShots)?(reqBody.OpponentsShots):[];
+        if (this.MyShots){
+            for (let i:number = 0; i<this.MyShots.length; i++){
+                let row:number = GameState.letterToNumber[this.MyShots[i].Position.Row];
+                let column:number = this.MyShots[i].Position.Column - 1
+                this.board[row][column] = (this.MyShots[i].WasHit)? 1 : -1;
+            }
+        }
+    }
+
+    public display():void{
+        console.log();
+        console.log(this.board);
+        console.log();
+    }
+
+    public isValidTarget(position:Position):boolean{
+        if (position.row >= 0){
+            if (position.row < 10){
+                if (position.column >= 0){
+                    if (position.column < 10){
+                        return (this.board[position.row][position.column]==0);
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public eliminateSunkenShips():void{
+        // first scans rows for sunken ships
+        for (let row:number =0; row<10; row++){
+            for (let column:number = 0; column<10; column++){
+                let startColumn:number = column;
+                if (this.board[row][column]==1){
+                    while ((column<10) && (this.board[row][column]==1)){
+                        column++;
+                    }
+                    let sunken:boolean = this.isSunken(row,startColumn,row,column-1);
+                    if (sunken){
+                        let shipLength:number = column-startColumn;
+                        this.remainingShips = this.remove(shipLength);
+                        this.sink(row,startColumn,row,column-1);
+                    }
+                }
+            }
+        }
+        // then checks columns for sunken ships
+        for (let column:number =0; column<10; column++){
+            for (let row:number = 0; row<10; row++){
+                let startRow:number = row;
+                if (this.board[row][column]==1){
+                    while ((row<10) && (this.board[row][column]==1)){
+                        row++;
+                    }
+                    let sunken:boolean = this.isSunken(startRow,column,row-1,column);
+                    if (sunken){
+                        let shipLength:number = row-startRow;
+                        this.remainingShips = this.remove(shipLength);
+                        this.sink(startRow,column,row-1,column);
+                    }
+                }
+            }
+        }
+    }
+
+    public remove(shipLength:number):number[]{
```

Comment:
Not a great name. What are we removing? When I see `GameState.remove` I can only assume that we're removing or deleting the gamestate, which isn't what we're doing.

Can you think of a clearer name?